packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -3864,27 +3864,41 @@ export class ContainerRuntime
 	public async getPendingLocalState(props?: {
 		notifyImminentClosure: boolean;
 	}): Promise<unknown> {
-		this.verifyNotClosed();
-		const waitBlobsToAttach = props?.notifyImminentClosure;
-		if (this._orderSequentiallyCalls !== 0) {
-			throw new UsageError("can't get state during orderSequentially");
-		}
-		const pendingAttachmentBlobs = await this.blobManager.getPendingBlobs(waitBlobsToAttach);
-
-		if (!pendingAttachmentBlobs && !this.hasPendingMessages()) {
-			return; // no pending state to save
-		}
+		return PerformanceEvent.timedExecAsync(
+			this.mc.logger,
+			{ eventName: "runtime_getPendingLocalState" },
+			async () => {
+				this.verifyNotClosed();
+				const waitBlobsToAttach = props?.notifyImminentClosure;
+				if (this._orderSequentiallyCalls !== 0) {
+					throw new UsageError("can't get state during orderSequentially");
+				}
+				const pendingAttachmentBlobs = await this.blobManager.getPendingBlobs(
+					waitBlobsToAttach,
+				);
+				const pending = this.pendingStateManager.getLocalState();
+				this.mc.logger.sendTelemetryEvent({
```

Comment:
same as the other, just add them to the event, rather than creating a new one

Reply:
These values are not available at the event level.

Reply:
at least attachmentBlobsSize and pendingOpsSize

Reply:
see above comment